app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductsDownloadFragment.java

```diff
@@ -0,0 +1,273 @@
+                eStatus.setText(getString(R.string.txtSaved));
+                eExtractButton.setImageResource(R.drawable.baseline_autorenew_24);
+            } else {
+                //data is not extracted
+                eStatus.setText(getString(R.string.txtNotSaved));
+                eExtractButton.setImageResource(R.drawable.baseline_save_alt_24);
+            }
+            dStatus.setVisibility(View.VISIBLE);
+            dStatus.setText(getString(R.string.txtDownloaded));
+        } else {
+            //data is not downloaded
+            dDownloadButton.setImageResource(R.drawable.ic_file_download_black_24dp);
+            dStatus.setVisibility(View.VISIBLE);
+            dStatus.setText(getString(R.string.txtNotDownloaded));
+            extractLayout.setVisibility(View.GONE);
+            eStatus.setVisibility(View.GONE);
+            dDeleteButton.setVisibility(View.GONE);
+        }
+
+        if (isDownloadOfflineProductServiceRunning) {
+            dProgressBar.setVisibility(View.VISIBLE);
+            dStatus.setVisibility(View.GONE);
+            dProgressBar.setIndeterminate(true);
+        } else if (isExtractOfflineProductServiceRunning) {
+            eProgressBar.setVisibility(View.VISIBLE);
+            eStatus.setVisibility(View.GONE);
+            eProgressBar.setIndeterminate(true);
+        }
+
+    }
+
+    @OnClick(R.id.d_download_button)
+    public void download(View v) {
+        if (Utils.isStoragePermissionGranted(getActivity())) {
+            if (DownloadOfflineProductService.isDownloadOfflineProductServiceRunning) {
+                Toast.makeText(getContext(), getString(R.string.toast_already_running), Toast.LENGTH_SHORT).show();
+            } else if (ExtractOfflineProductService.isExtractOfflineProductServiceRunning) {
+                Toast.makeText(getContext(), "Extraction in progress", Toast.LENGTH_SHORT).show();
+            } else {
+                Toast.makeText(getContext(), getString(R.string.toast_starting_download), Toast.LENGTH_SHORT).show();
+                Intent serviceIntent = new Intent(getContext(), DownloadOfflineProductService.class);
+                Objects.requireNonNull(getActivity()).startService(serviceIntent);
+                dProgressBar.setVisibility(View.VISIBLE);
+                dStatus.setVisibility(View.GONE);
+                dProgressBar.setIndeterminate(true);
+            }
+        }
+    }
+
+    @OnClick(R.id.d_delete_button)
+    public void delete(View v) {
+        if (isDownloadOfflineProductServiceRunning || isExtractOfflineProductServiceRunning) {
+            Toast.makeText(getContext(), getString(R.string.toast_already_running), Toast.LENGTH_SHORT).show();
+        } else {
+            if (settings.getBoolean("is_data_downloaded", false)) {
+                File file = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS) + File.separator + "fr.openfoodfacts.org.products.small.zip");
```

Comment:
Can you turn that into a non-translatable string so that it can work with Open Beauty Facts and the other variants ?